[feat][elasticsearch-sink] Option to strip out non printable characters (#15431)
apache/pulsar: site2/docs/io-elasticsearch-sink.md

| `compatibilityMode` | enum (AUTO,ELASTICSEARCH,ELASTICSEARCH_7,OPENSEARCH) | AUTO |  | Specify compatibility mode with the ElasticSearch cluster. `AUTO` value will try to auto detect the correct compatibility mode to use. Use `ELASTICSEARCH_7` if the target cluster is running ElasticSearch 7 or prior. Use `ELASTICSEARCH` if the target cluster is running ElasticSearch 8 or higher. Use `OPENSEARCH` if the target cluster is running OpenSearch. |
| `token` | String| false | " " (empty string)|The token used by the connector to connect to the ElasticSearch cluster. Only one between basic/token/apiKey authentication mode must be configured. |
| `apiKey` | String| false | " " (empty string)|The apiKey used by the connector to connect to the ElasticSearch cluster. Only one between basic/token/apiKey authentication mode must be configured. |

> | `stripNonPrintableCharacters` | Boolean| false | true|If `stripNonPrintableCharacters` is true, all non-printable characters are removed from the document. |

Reviewer: ```suggestion
| `stripNonPrintableCharacters` | Boolean| false | true| Whether to remove all non-printable characters from the document or not. If it is set to true, all non-printable characters are removed from the document. |
```

> | `stripNonPrintableCharacters` | Boolean| false | true|If stripNonPrintableCharacters is true, all non-printable characters will be removed from the document. |

Reviewer 2: ```suggestion
| `stripNonPrintableCharacters` | Boolean| false | true|If `stripNonPrintableCharacters` is true, all non-printable characters are removed from the document. |
```

Write in the simple present tense as much as possible if you are covering facts that were, are, and forever shall be true.

https://docs.google.com/document/d/1lc5j4RtuLIzlEYCBo97AC8-U_3Erzs_lxpkDuseU0n4/edit#bookmark=id.e8uqh1awkcnp
